src/env/as_env.rs

```diff
@@ -1,12 +1,14 @@
 use crate::{types::Interface, GasCosts};
-use std::collections::HashMap;
+use parking_lot::RwLock;
+use std::{collections::HashMap, sync::Arc};
 use wasmer::Global;
 
 use super::Metered;
 
 #[derive(Clone)]
 pub struct ASEnv {
     ffi_env: as_ffi_bindings::Env,
+    pub abi_enabled: Arc<RwLock<bool>>,
```

Comment:
Add doc here to explain that it is used to prevent abi call during wasm start func?

Reply:
I documented the entire ASEnv object

Reply:
why not use https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html which is much simpler and faster ?

Reply:
Also the original remark was treated please close this if you are ok with your question answer @damip